js/checkForNewRound.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Checks if there is a new round is by calling checkRoundStatus.php
+ * 
+ * Requires jQuery
+ */
+(function checkForNewRound() {
+    $.get('actions/pollRoundStatus.php', (result, status) => {
```

Comment:
Min uppfattning är att success anropas bara ifall man får en 200 från servern.

Jag är inte säker men detta kan innebära att den slutar fungera ifall servern inte kan svara 200 av någon anledning och därefter kommer den bara stå att den väntar på nästa fast aldrig uppdatera. Vill du verifiera detta på enklaste sätt genom att bara starta server, ladda sida sen stänga den och kolla ifall requests fortfarande görs.

Reply:
Du hade rätt, den stannar om den inte får en 200. Jag lägger att den bara kör vidare om den inte får rätt status.

Reply:
Okej så jag har typ löst detta, eller ja den börjar om iaf. Däremot kan det ju vara så att servern fortsätter ge andra statuskoder än 200 och inte notifiera användaren. Dock borde ju detta inte vara ett problem när man sitter där, men borde den ge en alert?

Reply:
Jag hade gjort alternativ 2 med att tvinga användaren att ladda om. Då får de fram vad som kan vara fel t.ex. 5XX, 404 (av någon anledning) eller att webbläsaren säger till "hej du har inget internet".

Gör inte en alert dock! Det kan vara irriterande att få popups särskilt om fliken är i bakgrunden. Se till att bara ändra texten  på sidan och lägg en knapp som laddar om sidan.

QOL förbättring du kan ha är att spara personliga koden som en variabel så den kan återanvändas under nästa röstning.

Reply:
Så, nu har jag lagt till en knapp och lite HTML/CSS för det. Jag kör ändå alert om man är i en omröstning och förlorar connection med servern (ja alerts är jävliga men om man sitter utan internet, skriver in sin kod och får error då är värre. Också lite onödigt med knappen där känner jag.

Kollade för `pollRoundStatus.php` och det funkade inte att bara echoa `$evote->ongoingRound()` :/

QOL får bli en ny pull request.

Det är just nu 5 s mellan varje poll, vet ej om man ska köra aggressivare?

Reply:
Huh, man hade ju hoppats på att den internt skulle anropa en toString som alla vettiga språk gör... Finns ternary som är kort och koncist `echo $evote->ongoingRound() ? 'true' : 'false';`

Yes kan vara mer viktigt att veta att den nuvarande omröstningen inte funkar av någon anledning.

Ah ser nu vad du har gjort med countdown. `cd` är inte ett lätt variabelnamn!

För min del kan du merga när allt funkar som det ska.